spec/load_test.rb

```diff
@@ -48,14 +49,14 @@ def timer
       end
     end
 
-    tournament.players.each do |player|
-      if player.opponents.uniq.length != player.pairings.count
-        puts "Player #{player.name} (#{player.active? ? :active : :dropped}) had #{player.opponents.uniq.length}/#{player.pairings.count} unique opponents:"
-        player.pairings.each do |pairing|
-          opp = pairing.opponent_for(player)
-          puts "\t#{pairing.round.number}: #{opp.name}"
-        end
-      end
-    end
+    # tournament.players.each do |player|
```

Comment:
TL;DR: This section warns me during a benchmark test if there were erroneous repeat matchups, but that wasn't relevant when I was testing single-sided (because repeat matchups are allowed)

This spec file is not a real spec file really (it doesn't run when you run `rspec` nor during builds since it's slow to run) but I've used it in the past to benchmark performance for tournaments with a lot of players and over many rounds.
As well as checking performance in the main section, I wrote this bottom section to validate after running that players didn't have repeat pairings. The BigSwiss strategy (which this was mostly written for) fudges the pairings at the end if things go wrong so I wanted to see how often it gave up and just paired randomly, resulting in repeats.
I commented it out here because single-sided obviously allows for repeat matchups so there were a lot of false positives. I suppose I could have rewritten it to check for third matchups, but I don't think that's really an issue; exhausting all possible matchups is basically impossible when you're allowed to play everyone twice.
I also suppose I could have not committed my changes to this file, it's just a helper script that might be useful to developers really.

Reply:
Ah, gotcha.  Maybe make it a rake task locked to the TEST env?  how does it not run when rspec is invoked?

Reply:
By default `rspec` runs only files that end with `_spec.rb`, so it skips this file when running the whole test suite. You can explicitly run the file with `rspec spec/load_test.rb` which is what I do if I want to run the load test.
I created it as a spec file because I like using the RSpec DSL to create the fixtures etc. I find it a little cleaner. Also, it IS a test, albeit not one I want to run every time the whole suite runs. You could create a rake task that does a similar thing, but you'd have to make sure it only runs in a test db (as you say) and create the objects directly (or do some work to allow RSpec syntax). I'm not strongly opposed to such a change, but I'm not eager to do it myself.

Reply:
Ah! That _spec.rb was the convention I was missing. Yeah, this all tracks. No need to move this file or anything, but please either add an arg/conditional or delete the commented code